### DISCUSSION TITLE
Action hook parity between Archive Products and Loop Grid elements?

### DISCUSSION THREAD

**Discussion by antonywalton** [elementor/elementor#27607] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

I am creating a Product Category Archive in Elementor Pro with the ‘Hello Elementor’ theme and WooCommerce. I have noticed that the action hooks that are fired on the loop for the archive template are different, per the element used to show the posts loop. This causes issues with plugin integration and code execution. In the archive template I have tried two elements:

- **Loop Grid** – This doesn’t seem to include any WC hooks, I have added all WC archive elements to the Loop Item (Add to Basket, Stock, Price, etc.) but this still does not fire any additional WC hooks
- **Archive Products** – This element does work with WC action hooks, so the correct hooks are being fired for additional functionality to work.

Currently these are the hooks I have found that seem to be missing from the Loop Grid (there may be others):

- woocommerce_after_shop_loop_item
- woocommerce_template_loop_add_to_cart
- woocommerce_single_variation
- woocommerce_single_variation_add_to_cart_button

### Describe the solution you'd like.

Are the action hooks mentioned above called in the Loop Grid element with _any_ WC element? The Loop Grid (or possibly a Loop Item template) element would be enhanced with appropriate WooCommerce hooks. These hooks could be included when adding the WooCommerce 'Add to Basket,' 'Stock,' or similar WooCommerce-only loop elements. 

As an interim solution, is there a code shim to add the missing hooks to an existing hook in the Loop Grid or Loop Item?

### Describe alternatives you've considered.

The WC hooks seem to be correctly included when using the Archive Products element. This is less desirable as it is an older element and has far less layout flexibility when compared with the Loop Grid.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by milkmanmatty** [comment]
Wild that this is still a problem almost a year on, does anyone know a workaround for this issue?

For woocommerce_after_shop_loop_item I ended up creating a `woocommerce/loop/add-to-cart.php` template file which only contained `global $product; do_action( 'woocommerce_' . $product->get_type() . '_add_to_cart' );`
Truly some of the dirtiest code I've written in a while. 

For the other missing hooks I'm just going to wait for Elementor to provide a way forward.
